invenio_administration/assets/semantic-ui/js/invenio_administration/details/DetailsComponent.js

```diff
@@ -0,0 +1,69 @@
+// This file is part of InvenioAdministration
+// Copyright (C) 2022 CERN.
+//
+// Invenio RDM is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import PropTypes from "prop-types";
+import React, { Component, useContext } from "react";
+import Overridable from "react-overridable";
+import { Table } from "semantic-ui-react";
+import AdminDetailsContext from "./AdminDetailsContext";
+
+function DetailsComponent({ ...props }) {
+  const ctx = useContext(AdminDetailsContext);
+  return <Element data={ctx.data} {...props} />;
+}
+
+const Element = ({ ...props }) => {
+  return (
+    <Overridable id="DetailsComponent.table">
+      <DetailsTable {...props} />
+    </Overridable>
+  );
+};
+
+Element.propTypes = {
+  columns: PropTypes.array.isRequired,
+  data: PropTypes.object.isRequired,
+};
+
+export class DetailsTable extends Component {
+  generateTableRow(key, pair) {
+    return (
+      <Table.Row key={key}>
+        <Table.Cell>
+          <b>{key}</b>
+        </Table.Cell>
+        <Table.Cell>{pair}</Table.Cell>
+      </Table.Row>
+    );
+  }
+
+  render() {
+    const { columns, data } = this.props;
+    let tableRows = [];
+    columns.map((col) => {
+      let key = col.key;
+      let text = col.text;
+      let value = data[key];
+      if (text && value && typeof value !== "object") {
```

Comment:
This does not allow for arrays or objects to be rendered in the table.
We can have `JSON.stringify()` but that would remove empty values, thus losing some information. 

In the future we might consider something better for this, for now it's just defensive programming.